hw/ip/otp_ctrl/data/otp_ctrl_testplan.hjson

```diff
@@ -60,52 +61,63 @@
+            - keep writing to OTP memory via DAI interface without asserting reset
+            - if digests are not locked, backdoor inject ECC correctable or uncorrectable errors
+            - issue reset and power initialization
+            - if the injected errors are all correctable errors, disable the `lc_bypass_chk_en`
+              after LC program request to create an LC partition check failure
 
-            - Check if the corresponding alerts are triggered
-            - Check if the error_code register is set correctly
-            Note that due to limited simulation time, for background checks, this test only write
-            random value that is less than 'd20 to the check period.
+            If fatal error is triggered, this test will check:
+            - OTP initialization failure triggers fatal alert
+            - `status`, `intr_state`, `err_code` CSRs reflect correct fatal error
+
+            If OTP initialization finished without any fatal error, this test will check:
+            - OTP initialization finishes with power init output goes to 1
+            - `status`, `intr_state`, `err_code` CSRs reflect ECC correctable error
             '''
       milestone: V2
-      tests: ["otp_ctrl_check_fail", "otp_ctrl_background_chks"]
+      tests: ["otp_ctrl_init_fail"]
     }
     {
-      name: init_fail
+      name: partition_check
       desc: '''
-            This test is based on OTP_CTRL smoke test and creats OTP_CTRL's initialization failure:
-            - Write and read OTP memory via DAI interface
-            - Randomly issue DAI digest command to lock HW partitions
-            - Keep writing to OTP memory via DAI interface
-            - If digests are not locked, backdoor inject ECC correctable or uncorrectable errors
-            - Issue reset and power initialization
+            Randomly program the partition check related CSRs including:
+            - `check_timeout`
+            - `integrity_check_period`
+            - `consistency_check_period`
+            - `check_trigger`
 
-            If fatal alert is triggered, this test will check:
-            - Otp_initialization failure triggers fatal alert
-            - Status register reflects the correct error
-            - Otp_ctrl's power init output stays 0
+            Create a failure scenario by randomly picking one of these three methods:
+            - inject ECC errors into the OTP macro via backdoor
+            - set the `check_timeout` CSR with a very small value
+            - write to a random OTP partition after digest is issued but before reset is asserted
 
-            If OTP init finished without fatal alert, this test will check:
-            - OTP init finishes with power init output goes to 1
-            - Status and interrupt registers reflect the correct ecc correctable error
+            **Checks**:
+            - check if the corresponding alerts are triggered
```

Comment:
Remove "check if", since this is a list of checks.

Reply:
Thanks Matute. Yup, I removed the check if.